### ISSUE TITLE
Cypress 4.2.0 10 low vulnerabilities concerning upgrade to minimist 1.2.3

### ISSUE THREAD

**Issue by Phillipe-Bojorquez** [cypress-io/cypress#6793]
### Current behavior:

Upgrading to cypress 4.2.1 by using `npm install --save-dev cypress@4.2.0` installs correctly but shows the following error:

```
found 10 low severity vulnerabilities
  run `npm audit fix` to fix them, or `npm audit` for details
```

Running `npm audit fix` takes no action even when using the force flag.  Looking at npm audit I see that there are sub dependencies for other libraries and a link to this issue: https://www.npmjs.com/advisories/1179

```
>         === npm audit security report ===                        
                                                                                
┌──────────────────────────────────────────────────────────────────────────────┐
│                                Manual Review                                 │
│            Some vulnerabilities require your attention to resolve            │
│                                                                              │
│         Visit https://go.npm.me/audit-guide for additional guidance          │
└──────────────────────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Low           │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ minimist                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=0.2.1 <1.0.0 || >=1.2.3                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ http-server                                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ http-server > optimist > minimist                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1179                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Low           │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ minimist                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=0.2.1 <1.0.0 || >=1.2.3                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ http-server                                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ http-server > portfinder > mkdirp > minimist                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1179                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Low           │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ minimist                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=0.2.1 <1.0.0 || >=1.2.3                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ wd                                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ wd > mkdirp > minimist                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1179                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Low           │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ minimist                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=0.2.1 <1.0.0 || >=1.2.3                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ cypress [dev]                                                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ cypress > extract-zip > mkdirp > minimist                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1179                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Low           │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ minimist                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=0.2.1 <1.0.0 || >=1.2.3                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ mocha [dev]                                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ mocha > mkdirp > minimist                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1179                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Low           │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ minimist                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=0.2.1 <1.0.0 || >=1.2.3                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ npm                                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ npm > libnpx > update-notifier > latest-version >            │
│               │ package-json > registry-auth-token > rc > minimist           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1179                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Low           │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ minimist                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=0.2.1 <1.0.0 || >=1.2.3                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ npm                                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ npm > update-notifier > latest-version > package-json >      │
│               │ registry-auth-token > rc > minimist                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1179                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Low           │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ minimist                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=0.2.1 <1.0.0 || >=1.2.3                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ npm                                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ npm > libnpx > update-notifier > latest-version >            │
│               │ package-json > registry-url > rc > minimist                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1179                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Low           │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ minimist                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=0.2.1 <1.0.0 || >=1.2.3                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ npm                                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ npm > update-notifier > latest-version > package-json >      │
│               │ registry-url > rc > minimist                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1179                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Low           │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ minimist                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=0.2.1 <1.0.0 || >=1.2.3                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ cypress [dev]                                                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ cypress > minimist                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1179                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
found 10 low severity vulnerabilities in 12869 scanned packages
  10 vulnerabilities require manual review. See the full report for details.
> 
```

### Desired behavior:
The desired behavior is that npm updates to cypress@4.2.0 will not have the current vulnerabilities.

### Versions

Mac OS 10.15.3 / npm v 6.13.4 / v10.19.0

**Comment by heitorlessa** [comment]
Sent PR for `mkdirp` maintainer since the remaining vulns are coming from that

https://github.com/isaacs/node-mkdirp/pull/13

**Comment by heitorlessa** [comment]
Ignore that, didn't realize majority of NPM package.json don't follow semver ranges, so that's gonna take a while -- `Mkdirp` has a major version that no longer depends on `minimist`, however that will require upstream changes either way

**Comment by Phillipe-Bojorquez** [comment]
http-server seems to be another major one.  Its on `optimist": "~0.6.1"`
https://github.com/http-party/http-server/blob/master/package.json#L88

**Comment by jennifer-shehane** [comment]
`minimist` has already been updated and the update will be released in the next version: https://github.com/cypress-io/cypress/pull/6747 I can work on getting the other deps that include minimist up to date.

**Comment by khitrenovich** [comment]
Another dependency - `extract-zip` - was just updated also (https://github.com/maxogden/extract-zip/pull/85).

**Comment by karlhorky** [comment]
Opened #6845 to address the `extract-zip` update.

**Comment by jennifer-shehane** [comment]
Just wanted to note that unless you’re hosting Cypress on a server and accepting unsanitized input from outside users - this vulnerability will not affect you. We think this is an *extremely unlikely* use case.

**Comment by khitrenovich** [comment]
> Just wanted to note that unless you’re hosting Cypress on a server and accepting unsanitized input from outside users - this vulnerability will not affect you. We think this is an _extremely unlikely_ use case.

It still turns our CI audit job red, though - and you don't want to keep it red for long time, otherwise people start ignoring it.

**Comment by jennifer-shehane** [comment]
This was released in [`4.3.0`](https://on.cypress.io/changelog#4-3-0).  Not sure why this wasn't picked up to be closed automatically. 

This comment thread has been locked. If you are still experiencing this issue after upgrading to
Cypress v4.3.0, please open a new issue..
